pkg/apis/pipeline/v1/pipeline_validation.go

```diff
@@ -470,5 +466,10 @@ func (ps *PipelineSpec) ValidateParamArrayIndex(ctx context.Context, params Para
 		arrayIndexParamRefs = append(arrayIndexParamRefs, extractArrayIndexingParamRefs(p)...)
 	}
 
+	// if there are array indexing param references, the api feature gate needs to be set to `alpha` or `beta`
+	if len(arrayIndexParamRefs) > 0 && !config.CheckAlphaOrBetaAPIFields(ctx) {
+		return fmt.Errorf(`indexing into array params: %v require "enable-api-fields" feature gate to be "alpha" or "beta"`, arrayIndexParamRefs)
```

Comment:
```suggestion
		return fmt.Errorf(`indexing into array params: %v requires "enable-api-fields" feature gate to be "alpha" or "beta"`, arrayIndexParamRefs)
```

Reply:
This will be a list of params in %v why change it to requires ?

Reply:
because "requires" refers to "indexing" not to "array params"

Reply:
maybe `invalid parameter expression %s: indexing into array params requires "enable-api-fields" feature gate to be "alpha" or "beta"`